cli/cmd/main.go

```diff
@@ -712,6 +793,70 @@ func DataCommand(c *cli.Context) error {
 	}
 
 	filter := &datapb.Filter{}
+	if err := createDataFilter(c, filter); err != nil {
+		return err
+	}
+
+	client, err := rdkcli.NewAppClient(c)
+	if err != nil {
+		return err
+	}
+
+	dataType := c.String(dataFlagDataType)
+	switch dataType {
+	case dataTypeBinary:
+		if err := client.BinaryData(c.String(dataFlagDestination), filter, c.Uint(dataFlagParallelDownloads)); err != nil {
+			return err
+		}
+	case dataTypeTabular:
+		if err := client.TabularData(c.String(dataFlagDestination), filter); err != nil {
+			return err
+		}
+	default:
+		return errors.Errorf("invalid data type %s", dataType)
+	}
+
+	return nil
+}
+
+// DeleteCommand runs the command for deleting data from the Viam cloud.
+func DeleteCommand(c *cli.Context) error {
+	if c.String(dataFlagDataType) != dataTypeBinary && c.String(dataFlagDataType) != dataTypeTabular {
```

Comment:
Can you pull this into some `validateDataType` that returns the below error since it's now being used in multiple places?

Reply:
Actually, with the switch below doing the same validation, I think this can be removed entirely. I think this error message is more informative though, so would suggest replacing that one with this one

Reply:
see above